util/Logger.cpp

```diff
@@ -318,6 +325,19 @@ void InitLoggingSystem(const std::string& log_file, const std::string& _unnamed_
     InfoLogger(log) << "Logger initialized at " << std::ctime(&date_time);
 }
 
+void ShutdownLoggingSystemFileSink() {
+    // The file sink may not be safe to use during static deinitialization, because of the
+    // following bug:
+
+    // http://www.boost.org/doc/libs/1_64_0/libs/log/doc/html/log/rationale/why_crash_on_term.html
+    // https://svn.boost.org/trac/boost/ticket/8642
+    // https://svn.boost.org/trac/boost/ticket/9119
+
+    // When either ticket is fixed the ShutdownLoggingSystem() function can be removed.
+
+    GetLoggersToSinkFrontEnds().ShutdownFileSinks();
```

Comment:
The file sink is actually, one backend coupled to a frontend per logger.  All of the frontends are shutdown, which shutsdown the single backend.  

The sinks then fall back to the `std::cout` default sink, which is why the logs that occur after the end of `main()` are dumped to the console.  Dumping the logs to the console is ugly, but better than burying potentially useful crash debugging log messages.

The human client does not also close the file sinks for the server or the AIs.  They do that on their own.